src/main/java/com/lothrazar/storagenetwork/item/ItemCollector.java

```diff
@@ -60,12 +59,16 @@ public void onEntityItemPickupEvent(EntityItemPickupEvent event) {
         BlockEntity tile = serverTargetWorld.getBlockEntity(dp.getBlockPos());
         if (tile instanceof TileMain) {
           TileMain network = (TileMain) tile;
-          //
-          int countUnmoved = network.insertStack(item, false);
+
+          // Create a new reference to the stack, try to insert that into the
+          // network, then change the original stack size so the player picks up
+          // only what remains, if anything.
+          int countUnmoved = network.insertStack(item.copy(), false);
+          item.setCount(countUnmoved);
+
+          // We still want to play the pickup sound, even if Minecraft silently
+          // deletes the stack we just emptied.
           if (countUnmoved == 0) {
```

Comment:
This handles the case in which the inventory network can hold the entire stack, but does not address a case in which the stack partially — but not completely — fits into the inventory, enabling a dupe exploit.